### PR TITLE
Modify convert op to convert from bf16 to float and vice-versa

### PR DESCRIPTION
**Pull request by nbpatel**
Current op only does regular casting. Modified it to do bf16 --> float and float --> bf16 conversion as well

### CODE REVIEW

File: src/ngraph/runtime/reference/convert.hpp [hunk 1]
```diff
@@ -41,6 +41,30 @@ namespace ngraph
                     out[i] = static_cast<char>(static_cast<bool>(arg[i]));
                 }
             }
+
+            template <typename TI>
+            void convert_float_to_bf16(void* arg, void* out, size_t count)
```
Comment (aprocter) [inline src/ngraph/runtime/reference/convert.hpp]: Does `convert<TI, ngraph::bfloat16>` not work? If not, could we find a way to make it work?

FWIW, we do have casts between bfloat16 and float32.

Comment (diyessi) [reply src/ngraph/runtime/reference/convert.hpp]: I think `static_cast` should be made to work in `bfloat16.hpp`

Comment (nbpatel) [reply src/ngraph/runtime/reference/convert.hpp]: ok, that seems to work...how about the other way round? do we support that as well?

Comment (aprocter) [reply src/ngraph/runtime/reference/convert.hpp]: It should already work, though it _might_ only work if you're going directly to/from `float`. (I dunno when/if/how the whole promotion machinery kicks in for "custom" numeric types in C++...)

Comment (aprocter) [reply src/ngraph/runtime/reference/convert.hpp]: I think there is an `operator float()` in `bfloat16`...

**Pull request merged**